yb-voyager/src/testutils/testing_utils.go

```diff
@@ -0,0 +1,357 @@
+	NotNull    bool    // Whether the column has a NOT NULL constraint
+	Default    *string // Default value, if any (nil means no default)
+}
+
+// Column represents a column's expected metadata
+type ColumnPropertiesPG struct {
+	DataType   string
+	IsPrimary  bool
+	IsNullable string
+	Default    interface{}
+}
+
+// CompareStructs compares two struct types and reports any mismatches.
+func CompareStructs(t *testing.T, actual, expected reflect.Type, structName string) {
+	assert.Equal(t, reflect.Struct, actual.Kind(), "%s: Actual type must be a struct. There is some breaking change!", structName)
+	assert.Equal(t, reflect.Struct, expected.Kind(), "%s: Expected type must be a struct. There is some breaking change!", structName)
+
+	for i := 0; i < max(actual.NumField(), expected.NumField()); i++ {
+		var actualField, expectedField reflect.StructField
+		var actualExists, expectedExists bool
+
+		if i < actual.NumField() {
+			actualField = actual.Field(i)
+			actualExists = true
+		}
+		if i < expected.NumField() {
+			expectedField = expected.Field(i)
+			expectedExists = true
+		}
+
+		// Assert field names match
+		if actualExists && expectedExists {
+			assert.Equal(t, expectedField.Name, actualField.Name, "%s: Field name mismatch at position %d. There is some breaking change!", structName, i)
+			assert.Equal(t, expectedField.Type.String(), actualField.Type.String(), "%s: Field type mismatch for field %s. There is some breaking change!", structName, expectedField.Name)
+			assert.Equal(t, expectedField.Tag, actualField.Tag, "%s: Field tag mismatch for field %s. There is some breaking change!", structName, expectedField.Name)
+		}
+
+		// Report missing or extra fields
+		if !actualExists && expectedExists {
+			t.Errorf("%s: Missing field %s of type %s. There is some breaking change!", structName, expectedField.Name, expectedField.Type)
+		}
+		if actualExists && !expectedExists {
+			t.Errorf("%s: Unexpected field %s of type %s. There is some breaking change!", structName, actualField.Name, actualField.Type)
+		}
+	}
+}
+
+// CompareJson compares two structs by marshalling them into JSON and reports any differences.
+func CompareJson(t *testing.T, outputFilePath string, expectedJSON string, exportDir string) {
+	// Read the output JSON file
+	outputBytes, err := os.ReadFile(outputFilePath)
+	if err != nil {
+		t.Fatalf("Failed to read output JSON file: %v", err)
+	}
+
+	// Compare the output JSON with the expected JSON
+	if diff := cmp.Diff(expectedJSON, string(outputBytes)); diff != "" {
+		t.Errorf("JSON file mismatch (-expected +actual):\n%s", diff)
+	}
+
+	// Can be used if we don't want to compare pretty printed JSON
+	// assert.JSONEqf(t, expectedJSON, string(outputBytes), "JSON file mismatch. There is some breaking change!")
+
+	// Remove the test directory
+	if err := os.RemoveAll(exportDir); err != nil {
+		t.Logf("Failed to remove test export directory: %v", err)
+	}
```

Comment:
lets do this in parent function, using defer

Reply:
Done